handlers/network.go

```diff
@@ -0,0 +1,75 @@
+package handlers
+
+import (
+	"encoding/json"
+	"eth2-exporter/db"
+	"eth2-exporter/rpc"
+	"eth2-exporter/services"
+	"eth2-exporter/types"
+	"eth2-exporter/utils"
+	"github.com/ethereum/go-ethereum/params"
+	"math/big"
+	"net/http"
+)
+
+const (
+	GWei = 1e9
+)
+
+// Supply godoc
+// @Summary Get total supply of all native network tokens in existence
+// @Tags Misc
+// @Description Returns information about total supply of all native network tokens in existence (in wei).
+// @Produce  json
+// @Success 200 {object} types.ApiResponse{data=types.SupplyResponse} "Success"
+// @Failure 400 {object} types.ApiResponse "Failure"
+// @Failure 500 {object} types.ApiResponse "Server Error"
+// @Router /api/v2/totalsupply [get]
+func Supply(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	genesisTotalSupply := utils.Config.Chain.GenesisTotalSupply
+
+	totalAmountWithdrawn, _, err := db.GetTotalAmountWithdrawn()
+	if err != nil {
+		logger.WithError(err).Error("error getting total amount withdrawn from db")
```

Comment:
Should this function continue if there is an error?
Seems like this could be added
```suggestion
		logger.WithError(err).Error("error getting total amount withdrawn from db")
		sendServerErrorResponse(w, r.URL.String(), "Appropriate message here")
		return
```
This also applies to the other errors.